provider/rackspace/provider_configurator.go

```diff
@@ -24,13 +26,30 @@ func (c *rackspaceConfigurator) ModifyRunServerOptions(options *nova.RunServerOp
 
 // GetCloudConfig implements ProviderConfigurator interface.
 func (c *rackspaceConfigurator) GetCloudConfig(args environs.StartInstanceParams) (cloudinit.CloudConfig, error) {
-	cloudcfg, err := cloudinit.New(args.Tools.OneSeries())
+	series := args.Tools.OneSeries()
+	cloudcfg, err := cloudinit.New(series)
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
 	// Additional package required for sshInstanceConfigurator, to save
 	// iptables state between restarts.
 	cloudcfg.AddPackage("iptables-persistent")
+
+	if args.InstanceConfig.EnableOSRefreshUpdate {
+		// cloud-init often fails to update APT caches
+		// during instance startup on RackSpace. Add an
+		// extra call to "apt-get update" with a sleep
+		// on failure to attempt to alleviate this.
+		// See lp:1677425.
+		os, err := jujuseries.GetOSFromSeries(series)
+		if err != nil {
```

Comment:
I don't think we should error here - if it's an unknown series, just assume not ubuntu and do nothing.
If the issue truly is rackspace only, here seems as good a place as any. It is a bit kludgy. Not sure off hand of anything better.

Reply:
OK, changed (but it will error in other places).